solutions/CF Party and Sweets.mdx

```diff
@@ -0,0 +1,114 @@
+---
+id: cf-1158A
+source: CF
+title: The Party and Sweets
+author: Jesse Choe
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/66993)
+
+Let's consider the following example:
+
+```
+2 2
+0 1
+1 0
+```
+
+The answer to this example is $-1$ since boy $2$ ends up giving girl $2$ too
+many sweets, even if boy $2$ gives girl $2$ the minimum number of sweets he
+could possibly give, which is $1$. However, girl $2$ received a maximum of no
+sweets, making it impossible.
+
+If a single boy gives more sweets than a girl received, then any arrangement of
+sweets is impossible with the constraints. More formally, if
+
+$$
+\max(b_1, b_2, \dots, b_{n-1}, b_n) > \min(g_1, g_2, \dots, g_{m-1}, g_m),
+$$
+
+then the answer is $-1$.
+
+If the input provided has an arrangement of sweets that follow the given
+constraints, then we can greedily find the answer.
+
+Since our objective is to minimize the total sum of sweets, it is optimal to
+make each sweet in the arrangement as small as possible. Since the smallest
+number of sweets a boy $i$ could give to a girl $j$ is $b_i$. Thus, the minimum number of sweets he could give out in total is $b_i \cdot m$. Thus, the answer to this problem is lower bounded by $\sum\limits_{i=1}^{n} b_i \cdot m$.
```

Comment:
This does not read well (I think I suggested it earlier, oops!)

```suggestion
Since our objective is to minimize the total number of sweets given out, let's first consider a lower bound on the answer. Since the fewest
number of sweets boy $i$ gives each girl is $b_i$, he gives out a total of at least $b_i \cdot m$ sweets. Thus, the answer to this problem is lower bounded by $\sum\limits_{i=1}^{n} b_i \cdot m$.
```